src/pfe/file-watcher/server/src/controllers/projectStatusController.ts

```diff
@@ -175,24 +175,24 @@ export async function updateStatus(req: IUpdateStatusParams): Promise<IUpdateSta
  * @param type <Required | String> - The status type. Supported types include: 'appState' and 'buildState'.
  * @param projectID <Required | String> - An alphanumeric identifier for a project.
  * @param status <Required | String> - The states for appState and buildState.
- * @param detailedAppStatus <Optional | String> - The detailed app status to update.
+ * @param detailedAppStatus <Optional | DetailedAppStatus> - The detailed app status to update.
  * @param msg <Required | String> - The new message to update.
  * @param appImageLastBuild <Optional | String> - The last app image timestamp.
  * @param buildImageLastBuild <Optional | String> - The last build image timestamp.
  * @param translatedMsg <Optional | String> - The translated new message.
  *
  * @returns Promise<void>
  */
-export async function updateProjectStatus(type: string, projectID: string, status: string, msg: string, appImageLastBuild?: string, buildImageLastBuild?: string, translatedMsg?: string, detailedAppStatus?: string): Promise<void> {
+export async function updateProjectStatus(type: string, projectID: string, status: string, msg: string, appImageLastBuild?: string, buildImageLastBuild?: string, translatedMsg?: string, detailedAppStatus?: DetailedAppStatus): Promise<void> {
 
     if (type == STATE_TYPES.appState) {
 
         const newState: string = status;
-        const newError = msg;
+        let newError = msg;
         const newDetailedState = detailedAppStatus;
         let oldState: AppState = AppState.unknown;
         let oldError = AppState.unknown;
-        let oldDetailedState = AppState.unknown;
+        let oldDetailedState = undefined;
```

Comment:
I would prefer to leave the default value as `undefined`. I do not want to send the non-sense empty object to ide; plus, `undefined` is easier to check(in case we need to) comparing with `detailedAppStatus.message.trim().length === 0`